pq-crypto/sike_r3/sikep434r3_kem.c

```diff
@@ -101,40 +102,22 @@ int sike_p434_r3_crypto_kem_dec(unsigned char *ss, const unsigned char *ct, cons
     EphemeralKeyGeneration_A(ephemeralsk_, c0_);
 
     // Verify ciphertext.
-    // If c0_ and ct are NOT equal, decaps failed and we overwrite the shared secret
-    // with pseudorandom noise (ss = H(s||ct)) by performing the copy (dont_copy = false).
-    //
-    // If c0_ and ct are equal, then decaps succeeded and we skip the overwrite and output
-    // the actual shared secret: ss = H(m||ct) (dont_copy = true).
-    int dont_copy = !(CRYPTO_memcmp(c0_, ct, SIKE_P434_R3_PUBLIC_KEY_BYTES));
-    constant_time_copy_or_dont(temp, sk, SIKE_P434_R3_MSG_BYTES, dont_copy);
+    // If selector = 0 then do ss = H(m||ct), else if selector = -1 load s to do ss = H(s||ct)
+    int8_t selector = CRYPTO_memcmp(c0_, ct, SIKE_P434_R3_PUBLIC_KEY_BYTES);
```

Comment:
You have to copy and use `ct_compare` function here (https://github.com/microsoft/PQCrypto-SIDH/blob/effa607f244768cdd38f930887076373604eaa78/src/fpx.c#L20) because `CRYPTO_memcmp` doesn't return -1 when the arrays are not equal (it returns some random value different from 0). And while you are at it, could you please move `ct_cmov` and `ct_compare` to `sikep434r3_fpx.c` because that's where the functions are in SIKE's repo.

Reply:
Done